packages/fxa-settings/src/models/integrations/oauth-integration.ts

```diff
@@ -2,14 +2,17 @@
  * License, v. 2.0. If a copy of the MPL was not distributed with this
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
 
+import AuthClient from 'fxa-auth-client/lib/client';
 import { Constants } from '../../lib/constants';
 import { IntegrationFlags } from '../../lib/integrations/interfaces/integration-flags';
+import { OAuthRelier } from '../reliers';
 
 import {
   BaseIntegration,
   IntegrationFeatures,
   IntegrationType,
 } from './base-integration';
+import { OAuthRedirectIntegration } from './oauth-redirect-integration';
```

Comment:
`auth_brokers/oauth-redirect` became `integrations/oauth-integration.ts`. I was kind of torn between naming it `OAuthRedirect` and just went with `OAuth` since it was shorter and there's not another kind of OAuth integration. Do we need this separate / new `oauth-redirect` or can we use `oauth-integration`?

Reply:
I set it up this way because I worry that over time the integration class will become catch all for everything, violate the single responsibility principle, and end up becoming a [god object](https://en.wikipedia.org/wiki/God_object). If the integration classes are designed just to corral integraiton logic that's fine, but having other classes / modules with single responsibilities seems ideal and generally will lead to better code quality metrics.

Reply:
> If the integration classes are designed just to corral integraiton logic that's fine

That's basically how I pictured them, I figured our logic in them would be limited as we could move "behaviors" into components themselves. I'm not sure if I see us having several OAuth integrations so it seems OK to me to have one, _but_ I'm fine with leaving it and revisiting later if this feels more right to you. Btw, I'd forgotten about the term "god object," thanks for linking lol.